backend/src/api/user.rs

```diff
@@ -36,8 +44,36 @@ pub async fn create_guest_user(
     auth_user: AuthenticatedUser,
     pool: web::Data<DbPool>,
 ) -> Result<HttpResponse, ServerError> {
-    // TODO random guest names
-    let user = User::new(&auth_user.uid, "Guest", true)?;
+    let user = User::new(&auth_user.uid, &random_guest_name(), true)?;
     user.insert(&pool).await?;
     Ok(HttpResponse::Created().json(user))
 }
+
+#[get("/user/{uid}/challenges")]
+pub async fn get_user_challenges(
+    uid: web::Path<String>,
+    auth_user: AuthenticatedUser,
+    pool: web::Data<DbPool>,
+) -> Result<HttpResponse, ServerError> {
+    auth_user.authorize(&uid)?;
+    let user = User::find_by_uid(pool.get_ref(), uid.as_ref()).await?;
+    let mut response: Vec<GameChallengeResponse> = Vec::new();
+    for challenge in &user.get_challenges(&pool).await? {
+        response.push(GameChallengeResponse::from_model_with_user(
+            challenge,
+            user.clone(),
+        )?);
+    }
+    Ok(HttpResponse::Ok().json(response))
+}
+
+#[get("/user/{uid}/games")]
+pub async fn get_user_games(
+    uid: web::Path<String>,
+    pool: web::Data<DbPool>,
+) -> Result<HttpResponse, ServerError> {
+    let _user = User::find_by_uid(pool.get_ref(), uid.as_ref()).await?;
+    // FIXME: actually return the user's games once that's implemented
```

Comment:
```suggestion
    // TODO @leex actually return the user's games once that's implemented
```

Reply:
I suspect we can merge this and once this is in I can/will finish the game api and then I have something to grep for